File: src/TicTacToe.java
import java.util.*;
//The computer dynamically adapts based off what you play
public class TicTacToe {
    static String pickWinner(String x, String y) {
        if (Objects.equals(x, "rock")) {
            if (Objects.equals(y, "rock")) {
                return "No one is";
            }
            if (Objects.equals(y, "paper")) {
                return "The computer is";
            }
            if (Objects.equals(y, "scissors")) {
                return "You are";
            }
        }
        if (Objects.equals(x, "paper")) {
            if (Objects.equals(y, "rock")) {
                return "You are";
            }
            if (Objects.equals(y, "paper")) {
                return "No one is";
            }
            if (Objects.equals(y, "scissors")) {
                return "The computer is";
            }
        }
        if (Objects.equals(x, "scissors")) {
            if (Objects.equals(y, "rock")) {
                return "The computer is";
            }
            if (Objects.equals(y, "paper")) {
                return "You are";
            }
            if (Objects.equals(y, "scissors")) {
                return "No one is";
            }
        }
        return "error message";
    }
    static String stabilizeChoice(String x) {
        if (Objects.equals(x, "1")) {
            return "rock";
        }
        if (Objects.equals(x, "2")) {
            return "paper";
        }
        if (Objects.equals(x, "3")) {
            return "scissors";
        }
        return "error message";
    }
    static List<String> aiOptions = new ArrayList<>();
    public static void addAntiChoice(String x) {
        if (Objects.equals(x, "rock")) {
            aiOptions.add("paper");
        }
        if (Objects.equals(x, "paper")) {
            aiOptions.add("scissors");
        }
        if (Objects.equals(x, "scissors")) {
            aiOptions.add("rock");
        }
    }
    public static void main(String[] args) throws InterruptedException {
        aiOptions.add("rock");
        aiOptions.add("paper");
        aiOptions.add("scissors");
        System.out.println("This Java program plays Tic-Tac-Toe");
        while (true) {
            System.out.print("\nChoose rock, paper, or scissors (or 1, 2, or 3)\n");
            Thread.sleep(500);
            Scanner myObj = new Scanner(System.in);
            String choice = myObj.nextLine();
            if (Objects.equals(choice, "1") || Objects.equals(choice, "2") || (Objects.equals(choice, "3"))) {
                choice = stabilizeChoice(choice);
            }
            addAntiChoice(choice);
            Random rand = new Random();
            int aiChoice = rand.nextInt(aiOptions.size() - 1);
            String stableaiChoice = aiOptions.get(aiChoice);
            System.out.print("Your choice is: ");
            System.out.println(choice);
            Thread.sleep(500);
            System.out.print("The computer's choice is: ");
            System.out.println(stableaiChoice);
            Thread.sleep(1000);
            String pickedWinner = pickWinner(choice, stableaiChoice);
            System.out.print(pickedWinner);
            System.out.println(" the winner");
            Thread.sleep(2000);
        }
    }
}

File: src/TicTacToe.kt
enum class Choices {
    ROCK, PAPER, SCISSORS
}
fun rotate(x: Choices): Choices {
    return when (x) {
        Choices.PAPER -> Choices.SCISSORS
        Choices.ROCK -> Choices.PAPER
        Choices.SCISSORS -> Choices.ROCK
    }
}
fun selectWinner(x: Choices, y:Choices): String {
    return when {
        (x==y) -> "No one is"
        (rotate(x)==y) -> "The computer is"
        else -> "You are"
    }
}
fun stabilizeChoice(x:String): String {
    return when (x) {
        "1" -> "rock"
        "2" -> "paper"
        "3" -> "scissors"
        else -> x
    }
}
fun main() {
    val aiOptions = mutableListOf("rock", "paper", "scissors")
    println("This Kotlin program plays Tic-Tac-Toe")
    while (true) {
        //Note: I couldn't find a way to make aiOptions.add work without the function being nested here
        fun addAntiChoice(x: String) {
            when(x) {
                "rock" -> aiOptions.add("paper")
                "paper" -> aiOptions.add("scissors")
                "scissors" -> aiOptions.add("rock")
            }
        }
        println("\nChoose rock, paper, or scissors (or 1, 2, or 3)")
        Thread.sleep(500)
        var choice2: String = readln()
        choice2 = stabilizeChoice(choice2)
        addAntiChoice(choice2)
        val choice = Choices.valueOf(choice2.uppercase())
        val aiChoice = (0..aiOptions.size-1).random()
        val stableaiChoice2 = aiOptions[aiChoice]
        val stableaiChoice = Choices.valueOf(stableaiChoice2.uppercase())
        print("Your choice is: ")
        println(choice.toString().lowercase())
        Thread.sleep(1000)
        print("The computer's choice is: ")
        println(stableaiChoice.toString().lowercase())
        Thread.sleep(1000)
        val pickedWinner = selectWinner(choice, stableaiChoice)
        print(pickedWinner)
        println(" the winner")
        Thread.sleep(2000)
    }
}

